### PR TITLE
[search] Do not emit private swimming pools.

### PR DESCRIPTION
**Pull request by vng**
Closes https://github.com/organicmaps/organicmaps/issues/7867
Closes https://github.com/organicmaps/organicmaps/issues/3269

### CODE REVIEW

File: drape_frontend/selection_shape_generator.cpp [hunk 1]
```diff
@@ -228,15 +226,9 @@ drape_ptr<RenderNode> SelectionShapeGenerator::GenerateSelectionGeometry(ref_ptr
   {
     mapDataProvider.ReadFeatures([&points](FeatureType & ft)
     {
-      if (ft.GetGeomType() != feature::GeomType::Line || points.size() > 1)
-        return;
-      points.reserve(5);
-      ft.ForEachPoint([&points](m2::PointD const & pt)
-      {
-       if (points.empty() || !points.back().EqualDxDy(pt, mercator::kPointEqualityEps))
```
Comment (biodranik) [inline drape_frontend/selection_shape_generator.cpp:236]: Weren't duplicate points filtered here? Do we have such cases in the data?

Does it make sense to write a test for it?

File: data/categories.txt [hunk 2]
```diff
@@ -8049,6 +8049,10 @@ sw:Bwawa la kuogelea
 fa:استخرشنا
 mr:जलतरण तलाव
 
+# Not searchable
+leisure-swimming_pool-private
+en:DoNotMatchThisType_OneLanguageDefinitionIsEnoughHere
```
[the remark this answers is not in the document]
Comment (vng) [reply data/categories.txt:8054]: This is not a hack. We do search subtypes when defining parent type. To concrete subtype, need to declare it manually.

File: drape_frontend/selection_shape_generator.cpp (hunk 1 above)
Comment (vng) [reply drape_frontend/selection_shape_generator.cpp:236]: Features already have unique points.

Comment (biodranik) [reply drape_frontend/selection_shape_generator.cpp:236]: ACK

File: data/categories.txt (hunk 2 above)
[the remark this answers is not in the document]
Comment (biodranik) [reply data/categories.txt:8054]: It looks like a hack. Does it make sense to document that behavior in categories.txt?

**Review by biodranik — APPROVED** [review]

File: data/categories.txt (hunk 2 above)
[the remark this answers is not in the document]
Comment (vng) [reply data/categories.txt:8054]: Comment about child-parent matching is already present at the beginning of this file.

**Pull request merged**